PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,880 @@
+
+    def validate_list(self, full_list, sub_list):
+        """Validate the sub list is within the full list.
+
+        :param full_list: full list
+        :param sub_list: sub list
+        :returns: Boolean
+        """
+        return True if all(elem in full_list for elem in sub_list) else False
+
+    def choose_subset_volumes(self, storagegroup, volume_list):
+        """Validate the sub list is within the full list.
+
+        :param storagegroup: full list
+        :param volume_list: sub list
+        :returns: volume_list
+        """
+        create_vol = False
+        self.smart_print('Here is the full list of volumes in SG %s: %s',
+                         DEBUG, storagegroup, volume_list)
+        txt = ('Which do you want to migrate (comma separated list): ',
+               volume_list)
+        txt_out = self.input(txt)
+        if txt_out:
+            sub_volume_list = txt_out.split(',')
+            sub_volume_list = [x.strip(' \'') for x in sub_volume_list]
+            if self.validate_list(volume_list, sub_volume_list):
+                if len(volume_list) == len(sub_volume_list):
+                    create_vol = True
+                volume_list = sub_volume_list
+            else:
+                print_str = ('Unable to validate your list, '
+                             'no volumes will be migrated.')
+                self.smart_print(print_str, WARNING)
+                volume_list = []
+        else:
+            self.smart_print('You cannot input an empty list', DEBUG)
+            txt = 'Do you want to choose again Y/N or X:'
+            txt_out = self.input(txt)
+            if self.check_input(txt_out, 'Y'):
+                volume_list, create_vol = self.choose_subset_volumes(
+                    storagegroup, volume_list)
+            else:
+                sys.exit()
+        return volume_list, create_vol
+
+    def get_volume_list(self, storagegroup):
+        """Get the list of volumes from the storagegroup.
+
+        :param storagegroup: the storage group name
+        :returns: List, Boolean
+        """
+        create_vol = False
+        volume_list = self.conn.provisioning.get_vols_from_storagegroup(
+            storagegroup)
+        print_str = 'There are %d volume in storage group %s.'
+        self.smart_print(print_str, DEBUG, len(volume_list), storagegroup)
+        txt = ('Do you want to migrate all %s volumes: Y/N: '
+               % len(volume_list))
+        txt_out = self.input(txt)
+        if self.check_input(txt_out, 'Y'):
+            # Move list of devices from old to new masking view
+            print_str = ('Moving all volumes between source '
+                         'and target storage groups.')
+            self.smart_print(print_str, DEBUG)
+            create_vol = True
+        else:
+            volume_list, create_vol = self.choose_subset_volumes(
+                storagegroup, volume_list)
+
+        return volume_list, create_vol
+
+    def choose_sg(
```

Comment:
Method name isn't consistent with other methods in file, `sg `instead of `storage_group`